tests/foreman/api/test_hostgroup.py

```diff
@@ -1,9 +1,175 @@
+# -*- encoding: utf-8 -*-
 """Tests for the ``hostgroups`` paths."""
+from fauxfactory import gen_string
+from nailgun import client, entities, entity_fields
+from robottelo.common.constants import (
+    FAKE_0_PUPPET_REPO,
+    PUPPET_MODULE_NTP_PUPPETLABS,
+)
 from robottelo.common.decorators import stubbed
+from robottelo.common.helpers import get_data_file, get_server_credentials
 from robottelo.test import APITestCase
 # (too-many-public-methods) pylint:disable=R0904
 
 
+class HostGroupTestCase(APITestCase):
+    """Tests for host group entity."""
+
+    def test_bz_1107708(self):
+        """@Test: Host that created from HostGroup entity with PuppetClass
+        assigned to it should inherit such puppet class information under
+        'all_puppetclasses' field
+
+        @Feature: HostGroup and Host
+
+        @Assert: Host inherited 'all_puppetclasses' details from HostGroup that
+        was used for such Host create procedure
+
+        @BZ: 1107708
```

Comment:
Emm. Yes, that test will be definitely blocked by mentioned defect as we actually test that defect is not present in system.